src/Microsoft.Health.Fhir.Api/Features/Throttling/ThrottlingMiddleware.cs

```diff
@@ -82,7 +88,19 @@ public sealed class ThrottlingMiddleware : IAsyncDisposable, IDisposable
             }
 
             // snapshot the configuration values to reduce the number of instructions that need to execute in the lock.
-            _concurrentRequestLimit = configuration.ConcurrentRequestLimit;
+            if (_baseConfiguration["DataStore"].Equals(KnownDataStores.CosmosDb, StringComparison.OrdinalIgnoreCase))
+            {
+                _concurrentRequestLimit = configuration.ConcurrentRequestLimit == 0 ? (int)ThrottlingLimitDefault.Gen1 : Math.Max(configuration.ConcurrentRequestLimit, (int)ThrottlingLimitDefault.Gen1);
```

Comment:
Do we necessarily need to have the == 0 check? The Math.Max between the default and the config value should be enough. For example, if it is 0 then it would take the default value for gen1 instead.

Reply:
Change has been made.

Reply:
Also, wouldn't we want to check against the "_concurrentRequestLimit" instead of calling the configuration.ConcurrentRequestLimit? I saw a comment regarding minimizing checking of the configuration directly and this is one of those values that they take a snapshot of.